mobile/redux/features/linkingSlice.ts

```diff
@@ -0,0 +1,128 @@
+import { MakeTxResponse } from "@gnolang/gnonative";
+import { createAsyncThunk, createSlice } from "@reduxjs/toolkit";
+import * as Linking from 'expo-linking';
+import { ThunkExtra } from "redux/redux-provider";
+
+interface State {
+    linkingParsedURl: Linking.ParsedURL | undefined;
+    queryParams: Linking.QueryParams | undefined;
+    path: string | undefined;
+    hostname: string | undefined;
+}
+
+const initialState: State = {
+    linkingParsedURl: undefined,
+    queryParams: undefined,
+    path: undefined,
+    hostname: undefined,
+};
+
+export const hasParam = (param: string, queryParams: Linking.QueryParams | undefined): boolean => {
+    return Boolean(queryParams && queryParams[param] !== undefined);
+}
+
+export const requestLoginForGnokeyMobile = createAsyncThunk<boolean>("tx/requestLoginForGnokeyMobile", async () => {
+    console.log("requesting login for GnokeyMobile");
+    const callback = encodeURIComponent('tech.berty.dsocial:///login-callback');
+    return await Linking.openURL(`land.gno.gnokey://tologin?callback=${callback}`);
+})
+
+type MakeTxAndRedirectParams = {
+    postContent: string,
+    callerAddressBech32: string,
+};
+
+export const makeCallTxAndRedirectToSign = createAsyncThunk<MakeTxResponse, MakeTxAndRedirectParams, ThunkExtra>("tx/makeCallTxAndRedirectToSign", async (props, thunkAPI) => {
+    const { callerAddressBech32, postContent } = props;
+
+    const packagePath = "gno.land/r/berty/social";
+    const fnc = "PostMessage";
+    const args: Array<string> = [postContent];
+    const gasFee = "1000000ugnot";
+    const gasWanted = BigInt(10000000);
+
+    const res = await thunkAPI.dispatch(makeCallTx({packagePath, fnc, args, gasFee, gasWanted, callerAddressBech32 })).unwrap();
+
+    setTimeout(() => {
+        const params = [`tx=${encodeURIComponent(res.txJson)}`, `address=${callerAddressBech32}`, `client_name=dSocial`, `reason=Post a message`];
+        Linking.openURL('land.gno.gnokey://tosign?' + params.join('&'))
+    }, 500)
+
+    return res;
+})
+
+type MakeTxParams = {
```

Comment:
I suggest to rename this to `MakeCallTxParams` . Each transaction type has its own params. For example, [send](https://github.com/gnolang/gnonative/blob/4bb275d89768062b2281613d9ed9fd925ad00e88/expo/src/api/types.ts#L74-L81) is a different transaction type and doesn't have params like `fnc`.